recipes/oatpp/all/conanfile.py

```diff
@@ -0,0 +1,70 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class OatppConan(ConanFile):
+    name = "oatpp"
+    description = "Modern Web Framework for C++"
+    homepage = "https://github.com/oatpp/oatpp"
+    license = "Apache-2.0"
+    topics = ("conan", "oat++", "oatpp", "web-framework")
+    url = "https://github.com/conan-io/conan-center-index"
+    generators = "cmake"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    exports_sources = "CMakeLists.txt"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("oatpp-{0}".format(self.version), self._source_subfolder)
+
+    def configure(self):
+        if self.settings.os == "Windows" and self.options.shared:
+            raise ConanInvalidConfiguration("oatpp can not be built as shared library on Windows")
+
+        if self.settings.compiler == "gcc" and self.settings.compiler.version in ["4.8", "4.9"]:
+            raise ConanInvalidConfiguration("oatpp can not be built with gcc 4.8 or 4.9")
```

Comment:
```suggestion
        if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "5":
            raise ConanInvalidConfiguration("oatpp requires GCC >=5")
```

Reply:
Conan has some nifty APIs. Thanks for the suggestion.